iiod/iiod.c

```diff
@@ -362,9 +361,11 @@ int main(int argc, char **argv)
 	char err_str[1024];
 	void *xml_zstd;
 	size_t xml_zstd_len = 0;
+	uint16_t port = IIOD_PORT;
+	int64_t val;
```

Comment:
Your `val` should be of type `unsigned long`, since it's the return type of `strtoul`.
Note that `unsigned long` is 32-bits on 32-bit CPUs.